stellargraph/core/graph.py

```diff
@@ -129,23 +181,64 @@ def __init__(
         target_name=globalvar.TARGET_ATTR_NAME,
         node_features=None,
         dtype="float32",
+        nodes=None,
+        edges=None,
+        source_column=globalvar.SOURCE,
+        target_column=globalvar.TARGET,
```

Comment:
I've been leaving off documenting this, because it's not settled, and I wanted to get something landed before perfecting it: https://github.com/stellargraph/stellargraph/issues/723

In particular, the whole doc-string should be re-written to not talk about networkx.

Is doing this as a follow up task ok?